### PR TITLE
Feature/fix

### PR DESCRIPTION
**Pull request by andriikamaldinov1**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/JsDaddy/ngx-mask/blob/develop/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [x] Feature
- [x] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/app/options/options.component.html [hunk 1]
```diff
@@ -1,19 +1,19 @@
-<section class="content">
+<section class="content column bg-full-white">
     <div class="card" *ngFor="let tile of docs; index as i">
         <div
-            class="title row"
-            [ngStyle]="{ color: checkChoose(choose, tile.id) ? '#ffcc00' : 'black' }">
+            class="mr-title title row uppercase ls-25"
+            [ngClass]="checkChoose(choose, tile.id) ? 'yellow' : 'black'">
```
Comment (GlebChiz) [inline src/app/options/options.component.html:5]: use pipe instead of component method - checkChoose

File: src/app/shared/asset/is-empty.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+    name: 'isEmpty',
+    standalone: true,
+})
+export class IsEmptyPipe implements PipeTransform {
+    public transform(value: string | undefined): string {
```
Comment (GlebChiz) [inline src/app/shared/asset/is-empty.pipe.ts:8]: May be better - return value || 'Empty'?

File: src/app/app.component.ts [hunk 3]
```diff
@@ -1,55 +1,62 @@
...
+    public title = 'Ngx-Mask';
+    public subtitle = 'Angular plugin to make masks on form fields and html elements';
+    public chips = ['Angular', 'TypeScript', 'Web', 'Input', 'Pipe', 'Show-Masks'];
 
-    @ViewChildren('accordion', { read: ElementRef }) public accordion!: QueryList<ElementRef>;
-
-    @ViewChildren('panel', { read: ElementRef }) public panel!: QueryList<ElementRef>;
+    public ngOnInit(): void {
+        this.inputVal = {
+            docs: ComDocs,
```
Comment (xakeppok) [inline src/app/app.component.ts:55]: why you need ngOnInit for this?

File: src/app/header/header.component.html [hunk 4]
```diff
@@ -1,27 +1,39 @@
...
-            <a href="https://jsdaddy.github.io/ngx-copypaste/">
-                <button class="default-btn">Ngx-copypaste</button>
+                <img width="34" height="29" [ngSrc]="'logo' | asset" alt="jsdaddy" class="logo" />
             </a>
         </span>
-        <span>
-            <a class="loader" href="https://jsdaddy.github.io/ngx-loader-indicator/">
-                <button class="default-btn">Ngx-loader-indicator</button>
+    </div>
+    <div class="center-side row ai-c" *ngFor="let items of headerDoc">
```
Comment (xakeppok) [inline src/app/header/header.component.html:9]: use track by for ngFor. check everywhere

File: src/app/header/header.component.html [hunk 5]
```diff
@@ -1,27 +1,39 @@
...
+        <span class="mrl-20" [ngClass]="checkIsActive === items.link | color">
```
Comment (xakeppok) [inline src/app/header/header.component.html:10]: move this logic to pipe please

File: src/app/header/header.component.scss [hunk 6]
```diff
@@ -9,58 +9,51 @@
...
+            padding: 20px;
         }
     }
     .center-side {
-        display: flex;
-        align-items: center;
-        .default-btn, .active-btn {
+        .yellow {
+            button {
+                background: var(--black);
```
Comment (xakeppok) [inline src/app/header/header.component.scss:22]: you can use black and bg-black classes for this button

File: src/app/header/header.component.scss [hunk 7]
```diff
@@ -69,11 +62,99 @@
...
+                content: url('/assets/img/logo-white.svg');
+            }
+        }
+        .open-nav {
+            img {
+                width: 68px;
+                height: 68px;
+                position: relative;
+                top: 1px;
+                content: url('/assets/img/close.svg');
```
Comment (xakeppok) [inline src/app/header/header.component.scss:82]: is image changed only on mobile?

File: src/app/header/header.component.scss [hunk 8]
```diff
@@ -69,11 +62,99 @@
...
+    &.header {
+        display: flex !important;
+    }
+    &.nav-section {
+        display: none !important;
+    }
+}
+@media screen and (max-width: $desktopWidth) {
+    .left-side {
+        width: 100px !important;
```
Comment (xakeppok) [inline src/app/header/header.component.scss:101]: use the same nested css in this case you dont need important

File: src/app/header/header.component.scss [hunk 9]
```diff
@@ -69,11 +62,99 @@
...
+        .logo {
+            padding: 20px 0 0 10px !important;
+        }
+    }
+}
+
+@media screen and (max-width: $mobileWidth) {
```
Comment (xakeppok) [inline src/app/header/header.component.scss:108]: same here. and check it everywhere

File: src/app/header/header.component.ts [hunk 10]
```diff
@@ -1,12 +1,51 @@
...
+        NgClass,
+        AssetPipe,
+        ColorPipe,
+        HidePipe,
+        VisitBtnComponent,
+    ],
 })
-export class HeaderComponent {}
+export class HeaderComponent implements OnInit {
+    public headerDoc = [
```
Comment (xakeppok) [inline src/app/header/header.component.ts:25]: create interface fo this

File: src/app/header/header.component.ts [hunk 11]
```diff
@@ -1,12 +1,51 @@
...
+            link: 'https://jsdaddy.github.io/ngx-loader-indicator/',
+        },
+    ];
+
+    public showNav = false;
+
+    public checkIsActive!: string;
+
+    public ngOnInit(): void {
+        this.checkIsActive = window.location.href;
```
Comment (xakeppok) [inline src/app/header/header.component.ts:45]: can we do this without onInit?

File: src/app/header/header.component.ts [hunk 12]
```diff
@@ -1,12 +1,51 @@
...
+    }
+
+    public showNavBlock(): void {
```
Comment (xakeppok) [inline src/app/header/header.component.ts:48]: toggleNavBlock

File: src/app/options/options.component.html [hunk 13]
```diff
@@ -38,53 +38,64 @@
...
+        <div class="card-text column pd-15">
+            <span class="sub-text full-white title ls-25">Mask:</span>
+            <span class="green-view ai-start span">(000) 000-0000</span>
         </div>
-        <div class="card-text pd-15">
-            <div class="sub-text">Value:</div>
-            <span class="greenVIew">123456789</span>
+        <div class="card-text column pd-15">
+            <span class="sub-text full-white title ls-25">Value:</span>
+            <span class="green-view ai-start span">123456789</span>
```
Comment (xakeppok) [inline src/app/options/options.component.html:91]: is it make sense to create reusable component for all these card ?

File: src/app/options/options.component.scss [hunk 14]
```diff
@@ -1,176 +1,137 @@
...
+        .light-card {
+            flex: 100%;
+            margin-bottom: 10px;
+        }
+    }
+}
+
+@media screen and (min-width: $mobileWidth) and (max-width: $desktopWidth) {
+    .cards-block {
+        flex-direction: unset !important;
```
Comment (xakeppok) [inline src/app/options/options.component.scss:120]: please use correct nested styles. check everywhere

File: src/app/shared/accordion/accordion.component.html [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+                        class="active-svg"
+                        alt="" />
+                </div>
+                <button class="accordion-btn bg-black full-white pointer">
+                    {{ list.header }}
+                </button>
+            </div>
+            <div class="panel full-white text bg-black" #panel>
+                <p
+                    *ngFor="let item of list.text"
```
Comment (xakeppok) [inline src/app/shared/accordion/accordion.component.html:36]: and trackBY and use everywhere

File: src/app/shared/accordion/accordion.component.html [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+                    class="accordion-box pointer"
+                    [ngClass]="item.id === chosenItem | color"
+                    id="{{ item.id }}"
+                    (click)="handleClick(item.id, item.scrollTo)"
+                    [routerLink]="[]"
```
Comment (xakeppok) [inline src/app/shared/accordion/accordion.component.html:41]: what does [] means in routerLink?

Comment (xakeppok) [reply src/app/shared/accordion/accordion.component.html:41]: for what it was used thee?

File: src/app/shared/accordion/accordion.component.scss [hunk 17]
```diff
@@ -0,0 +1,191 @@
+@import 'src/styles';
+:host {
+    display: block;
+    height: 100%;
+}
+
+.show {
```
Comment (xakeppok) [inline src/app/shared/accordion/accordion.component.scss:7]: it should be in :host nested i guess

File: src/app/shared/accordion/accordion.component.ts [hunk 18]
```diff
@@ -0,0 +1,98 @@
...
+    public chosenList!: number;
+
+    @Input() public lists!: IListItem[];
+    @Output() public itemAccordion = new EventEmitter<number>();
+    @Output() public itemInAccordion = new EventEmitter<number>();
+    @ViewChildren('accordion', { read: ElementRef }) public accordion!: QueryList<ElementRef>;
+    @ViewChildren('panel', { read: ElementRef }) public panel!: QueryList<ElementRef>;
+
+    public ngOnInit(): void {
+        this.chosenItem = 1;
```
Comment (xakeppok) [inline src/app/shared/accordion/accordion.component.ts:49]: why are you using ngOnInit there?

Comment (xakeppok) [reply src/app/shared/accordion/accordion.component.ts:49]: public chosenItem =1
    public chosenList! = 1

File: src/app/shared/accordion/accordion.component.ts [hunk 19]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    public switchDoc(index: number): void {
+        this.itemAccordion.emit(index);
+    }
+
+    public handleClick(idItem: number, scrollTo: string | undefined): void {
+        this.chosenItem = idItem;
+        this.itemInAccordion.emit(idItem);
+        setTimeout(() => {
```
Comment (xakeppok) [inline src/app/shared/accordion/accordion.component.ts:74]: why are you using setTimeout?

File: src/app/shared/asset/color.pipe.ts [hunk 20]
```diff
@@ -0,0 +1,11 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+    name: 'color',
+    standalone: true,
+})
+export class ColorPipe implements PipeTransform {
```
Comment (xakeppok) [inline src/app/shared/asset/color.pipe.ts:7]: please write unit tests for all pipes

File: src/app/shared/accordion/accordion.component.ts [hunk 21]
```diff
@@ -0,0 +1,98 @@
...
+            }
+        });
+    }
+
+    public toggle(index: number): void {
+        this.accordion.get(index)?.nativeElement.classList.toggle('active');
+        const panel = this.panel.get(index)?.nativeElement;
+        panel.style.maxHeight = panel.style.maxHeight ? null : panel.scrollHeight + 'px';
+        const accordionsArray = this.accordion.toArray().map((el) => el.nativeElement.classList);
+        accordionsArray.map((el, i) => {
```
Comment (xakeppok) [inline src/app/shared/accordion/accordion.component.ts:90]: why are you using map if you are not returing anything?

File: src/app/shared/accordion/accordion.component.ts [hunk 22]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    public ngAfterViewInit(): void {
+        this.openFirstAccordion();
+    }
+
+    public openFirstAccordion(): void {
+        this.accordion.first.nativeElement.classList.toggle('active');
+        const panel = this.accordion.first.nativeElement.nextElementSibling;
+        panel.style.maxHeight = panel.style.maxHeight ? null : panel.scrollHeight + 'px';
```
Comment (xakeppok) [inline src/app/shared/accordion/accordion.component.ts:60]: i saw it in to places. maybe create method for this?

**Pull request closed without merging**